crates/tap-agent/src/tap/context.rs

```diff
@@ -152,36 +152,19 @@ impl NetworkVersion for Horizon {
 /// Context used by [tap_core::manager::Manager] that enables certain helper methods
 ///
 /// This context is implemented for PostgresSQL
-#[derive(Clone)]
+#[derive(Clone, bon::Builder)]
 pub struct TapAgentContext<T> {
     pgpool: PgPool,
+    #[cfg_attr(test, builder(default = crate::test::ALLOCATION_ID_0))]
     allocation_id: Address,
+    #[cfg_attr(test, builder(default = test_assets::TAP_SENDER.1))]
     sender: Address,
+    #[cfg_attr(test, builder(default = crate::test::INDEXER.1))]
     indexer_address: Address,
     escrow_accounts: Receiver<EscrowAccounts>,
     /// We use phantom data as a marker since it's
     /// only used to define what methods are available
     /// for each type of network
+    #[builder(default = PhantomData)]
```

Comment:
I recommend using [`#[builder(skip)]`](https://bon-rs.com/reference/builder/member/skip) for `PhantomData` fields to avoid generating the setters for this field at all.